test/Utilities/copy.jl

```diff
@@ -157,3 +157,70 @@ end
         @test dest.added_constrained[idxmap[vi].value]
     end
 end
+
+
+mutable struct OrderingConstrainedVariablesModel <: MOI.ModelLike
+    constraintIndices       ::Array{MOI.ConstraintIndex}
+    inner                   ::MOIU.Model{Float64}
+    OrderingConstrainedVariablesModel() = new(MOI.ConstraintIndex[], MOIU.Model{Float64}())
+end
+        
+
+
+MOI.add_variables(model::OrderingConstrainedVariablesModel, n) = MOI.add_variables(model.inner, n)
+MOI.add_variable(model::OrderingConstrainedVariablesModel) = MOI.add_variable(model.inner)
+
+function MOI.add_constraint(model::OrderingConstrainedVariablesModel, f::F, s::S) where {F<:MOI.AbstractFunction, S<:MOI.AbstractSet}
+    ci = MOI.add_constraint(model.inner, f, s)
+    push!(model.constraintIndices, ci)
+    return ci
+end
+
+function MOI.copy_to(dest::OrderingConstrainedVariablesModel, src::MOI.ModelLike; kws...)
+    MOIU.automatic_copy_to(dest, src; kws...)
+end
+
+MOIU.supports_default_copy_to(model::OrderingConstrainedVariablesModel, ::Bool) = true
+
+function MOI.empty!(model::OrderingConstrainedVariablesModel)
+    model.constraintIndices = MOI.ConstraintIndex[]
+    MOI.empty!(model.inner)
+end
+
+
+MOI.supports_constraint(::OrderingConstrainedVariablesModel, ::Type{MOI.VectorOfVariables}, ::Type{MOI.Nonnegatives}) = false
+MOI.supports_add_constrained_variables(::OrderingConstrainedVariablesModel, ::Type{MOI.Nonnegatives}) = true
+
+MOI.supports_constraint(::OrderingConstrainedVariablesModel, ::Type{MOI.VectorOfVariables}, ::Type{MOI.Nonnegatives}) = true
+MOI.supports_add_constrained_variables(::OrderingConstrainedVariablesModel, ::Type{MOI.Nonpositives}) = false
+
+MOI.supports_constraint(::OrderingConstrainedVariablesModel, ::Type{MOI.SingleVariable}, ::Type{<:MOI.GreaterThan}) = true
+MOI.supports_add_constrained_variable(::OrderingConstrainedVariablesModel, ::Type{<:MOI.GreaterThan}) = false
+
+MOI.supports_constraint(::OrderingConstrainedVariablesModel, ::Type{MOI.SingleVariable}, ::Type{<:MOI.LessThan}) = false
+MOI.supports_add_constrained_variable(::OrderingConstrainedVariablesModel, ::Type{<:MOI.LessThan}) = true
+
+
+@testset "Create variables using supports_add_constrained_variable(s) (#987)" begin
+    # With vectors
+    src = MOIU.Model{Float64}()
+    a, c1 = MOI.add_constrained_variables(src, MOI.Nonpositives(3))
+    c2 = MOI.add_constraint(src, a, MOI.Nonnegatives(3))
+    dest = OrderingConstrainedVariablesModel()
+
+    index_map = MOI.copy_to(dest, src)
+
+    @test typeof(c1) == typeof(dest.constraintIndices[2])
+    @test typeof(c2) == typeof(dest.constraintIndices[1])
```

Comment:
Could you try a model where the order is reversed as suggested in https://github.com/JuliaOpt/MathOptInterface.jl/pull/1101#discussion_r434667027 ? You can save most of the code between the models with an abstract types between the two as only the supports functions change?

Reply:
Is the new commit what you meant?

Reply:
Yes, exactly !